patches/net/minecraft/world/level/block/state/BlockBehaviour.java.patch

```diff
@@ -150,6 +157,23 @@
          BlockBehaviour.StatePredicate isRedstoneConductor = (p_284888_, p_284889_, p_284890_) -> p_284888_.isCollisionShapeFullBlock(p_284889_, p_284890_);
          BlockBehaviour.StatePredicate isSuffocating = (p_284885_, p_284886_, p_284887_) -> p_284885_.blocksMotion()
                  && p_284885_.isCollisionShapeFullBlock(p_284886_, p_284887_);
+@@ -974,6 +_,7 @@
+         FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
+         @Nullable
+         BlockBehaviour.OffsetFunction offsetFunction;
++        private it.unimi.dsi.fastutil.objects.ReferenceSet<net.neoforged.neoforge.flag.Flag> requiredFlags = it.unimi.dsi.fastutil.objects.ReferenceSets.emptySet();
+ 
+         private Properties() {
+         }
+@@ -1020,6 +_,8 @@
+             blockbehaviour$properties.offsetFunction = blockbehaviour$properties1.offsetFunction;
+             blockbehaviour$properties.spawnTerrainParticles = blockbehaviour$properties1.spawnTerrainParticles;
+             blockbehaviour$properties.requiredFeatures = blockbehaviour$properties1.requiredFeatures;
++            blockbehaviour$properties.requiredFlags = new it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet<>();
++            blockbehaviour$properties.requiredFlags.addAll(blockbehaviour$properties1.requiredFlags);
```

Comment:
Won't this produce an empty `ReferenceOpenHashSet` when copying a `Properties` object which had no required flags?

Reply:
oh yeah, if theres no flags on the thing we are copying it would still be mutable. I can fix that